parachain/pallets/ethereum-light-client/src/lib.rs

```diff
@@ -415,29 +457,38 @@ pub mod pallet {
 			Ok(())
 		}
 
+		fn find_finalized_decendant(
+			best_block_id: &EthereumHeaderId,
```

Comment:
You should probably just pass a `H256` instead of `EthereumHeaderId` here, since the function is only interested in the block hash.